lib/subscribers/google-genai/generate-content-stream.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2025 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const GoogleGenAIGenerateContentSubscriber = require('./generate-content')
+const { AI } = require('../../../lib/metrics/names')
+
+class GoogleGenAIGenerateContentStreamSubscriber extends GoogleGenAIGenerateContentSubscriber {
+  constructor({ agent, logger }) {
+    super({ agent, logger, channelName: 'nr_generateContentStreamInternal' })
+  }
+
+  handler(data, ctx) {
+    if (!this.agent?.config?.ai_monitoring?.streaming?.enabled) {
+      return ctx
+    }
+
+    return super.handler(data, ctx)
+  }
+
+  /**
+   * Instruments the streaming response by wrapping the next function.
+   * @param {object} params function params
+   * @param {object} params.request the original request object
+   * @param {object} params.response the original response object
+   * @param {TraceSegment} params.segment the active trace segment
+   * @param {Transaction} params.transaction the active transaction
+   */
+  instrumentStream({ request, response, segment, transaction }) {
+    const originalNext = response.next
+    let err
+    let content
+    let modelVersion
+    let finishReason
+    let entireMessage = ''
+    response.next = async function wrappedNext(...nextArgs) {
+      let result = {}
+      try {
+        result = await originalNext.apply(response, nextArgs)
+        if (result?.value?.text) {
+          modelVersion = result?.value?.modelVersion
+          content = result?.value?.candidates?.[0]?.content
+          entireMessage += result.value.text // readonly variable that equates to result.value.candidates[0].content.parts[0].text
+        }
+        if (result?.value?.candidates?.[0]?.finishReason) {
+          finishReason = result.value.candidates[0].finishReason
+        }
+      } catch (streamErr) {
+        err = streamErr
+        throw err
+      } finally {
+        // Update segment duration since we want to extend the
+        // time it took to handle the stream
+        segment.touch()
+
+        // result will be {value: undefined, done: true}
+        // when the stream is done, so we need to create
+        // a mock GenerateContentResponse object with
+        // the entire message
+        //
+        // also need to enter this block if there was an
+        // error, so we can record it
+        if (result?.done || err) {
+          if (content) {
+            content.parts[0].text = entireMessage
+            result.value = {
+              candidates: [
+                { content, finishReason }
+              ],
+              modelVersion
+            }
+          }
+
+          this.recordChatCompletionMessages({
+            segment,
+            transaction,
+            request,
+            response: result?.value,
+            err
+          })
+        }
+      }
+      return result
+    }.bind(this)
+  }
+
+  asyncEnd(data) {
+    if (!this.agent?.config?.ai_monitoring?.streaming?.enabled) {
```

Comment:
don't you want this to also go up in the handler? I know you'd still have to check if it was disabled to not wrap the stream but incrementing the metric may make more sense above